hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionImpl.java

```diff
@@ -384,22 +391,7 @@ private void forceCreateBackupLogs() {
             }
         }
 
-        for (Future future : futures) {
-            try {
-                future.get(timeoutMillis, MILLISECONDS);
-            } catch (MemberLeftException e) {
-                nodeEngine.getLogger(Transaction.class).warning("Member left while replicating tx begin: " + e);
-            } catch (Throwable e) {
-                if (e instanceof ExecutionException) {
-                    e = e.getCause() != null ? e.getCause() : e;
-                }
-                if (e instanceof TargetNotMemberException) {
-                    nodeEngine.getLogger(Transaction.class).warning("Member left while replicating tx begin: " + e);
-                } else {
-                    RETHROW_TRANSACTION_EXCEPTION.handleException(e);
-                }
-            }
-        }
+        waitWithDeadline(futures, timeoutMillis, MILLISECONDS, replicationTxExceptionHandler);
```

Comment:
there is a change in timeout - it user to be per-future, now the timeout is global for all futures - is this change intentional?